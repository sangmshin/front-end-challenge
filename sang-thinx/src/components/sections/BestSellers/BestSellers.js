import React from 'react';
import { array } from 'prop-types';
import cx from 'classnames';
import { ProductCard } from 'components/elements';
import { isArrayWithLength } from 'core/utils';
import styles from './BestSellers.module.scss';

export const BestSellers = ({ best_selling_products: bestSellers }) => {
  if (!isArrayWithLength(bestSellers)) return null;

  const productCards = bestSellers.map(product => (
    <ProductCard
      {...product}
      className={styles['best-seller-card']}
      key={product.product_title}
    />
  ));

  return (
    <section className={cx(styles.root, 'page-gutter')}>
      <div className={cx(styles['products-wrapper'], 'page-max-width')}>
        {productCards}
      </div>
    </section>
  )
};

BestSellers.propTypes = {
  best_selling_products: array
};

BestSellers.defaultProps = {
  best_selling_products: []
};

export default BestSellers;