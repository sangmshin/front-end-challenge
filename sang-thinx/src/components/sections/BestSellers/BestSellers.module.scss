@import 'variables';

.root {
  margin: 50px 0;

  .best-seller-card {
    margin-bottom: 45px;

    &:last-of-type {
      margin-bottom: 0px;
    }

    h2 {
      text-decoration: underline;
      font-size: 1rem;
      letter-spacing: 2px;
      font-weight: normal;
      padding: 10px 0 0;
    }

    p {
      font-size: 0.6875rem;
    }
  }
}

@media (min-width: $screen-md) {
  .root {
    max-width: 960px;
    margin: 50px auto;

    .products-wrapper {
      display: flex;
      flex-wrap: wrap;

      .best-seller-card {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        width: 50%;
        margin: 0 auto 60px;

        &:nth-of-type(3n+2) {
          align-items: flex-end;
        }

        img {
          max-height: 500px;
          max-width: 100%;
        }

        &:nth-child(3n+3) {
          width: 100%;

          img {
            max-width: 500px;
          }
        }

        h2 {
          padding: 20px 0 0;
        }
      }
    }
  }
}