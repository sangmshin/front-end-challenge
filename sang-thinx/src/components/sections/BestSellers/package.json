{
  "name": "BestSellers",
  "main": "./BestSellers.js"
}
