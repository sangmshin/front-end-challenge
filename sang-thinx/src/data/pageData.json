
{
  "template": "product-detail",
  "hero": {
    "images": [
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_01_1024x1024.jpg", "alt": "Hi-Waist front 1"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_02_1024x1024.jpg", "alt": "Hi-Waist front 2"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_03_1024x1024.jpg", "alt": "Hi-Waist front 3"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_front_04_1024x1024.jpg", "alt": "Hi-Waist front 4"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_side_1024x1024.jpg", "alt": "Hi-Waist side"
      },
      {
        "src": "https://thinx-assets.s3.amazonaws.com/highwaist_black_back_1024x1024.jpg", "alt": "Hi-Waist back"
      }
    ],
    "product_title": "Hi - Waist",
    "product_droplets": "Heavy days. Holds up to 2 tampon's worth",
    "description": "The phrase \"treat yo'self\" reaches new belly-button brushing heights with the Hi-Waist, and it's here just in time for cuffing season.",
    "price": "35.00",
    "available_colors": [
      "black",
      "beige"
    ],
    "available_sizes": [
      "xsmall",
      "small",
      "medium",
      "large",
      "xlarge",
      "xxlarge"
    ],
    "size_chart": true,
    "savings": {
      "label": "build a cycle set and save up to 20%",
      "url": ""
    }
  },
  "best_sellers": [
    {
      "product_title": "Hiphugger",
      "product_droplets": "heavy days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_1.jpg"
    },
    {
      "product_title": "Cheeky",
      "product_droplets": "light days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_2.jpg"
    },
    {
      "product_title": "Thong",
      "product_droplets": "lightest days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_3.jpg"
    },
    {
      "product_title": "Sport",
      "product_droplets": "medium days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_4.jpg"
    },
    {
      "product_title": "Boyshort",
      "product_droplets": "light days",
      "image": "https://thinx-assets.s3.amazonaws.com/best_seller_5.jpg"
    }
  ]
}
