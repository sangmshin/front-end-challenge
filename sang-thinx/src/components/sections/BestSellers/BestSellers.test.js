import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BestSellers } from './BestSellers';
import { best_sellers } from 'data/pageData.json';

describe('Component :: BestSellers', () => {
  describe('#render', () => {
    const wrapper = shallow(
      <BestSellers {...best_sellers} />
    );
    it('should match snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
