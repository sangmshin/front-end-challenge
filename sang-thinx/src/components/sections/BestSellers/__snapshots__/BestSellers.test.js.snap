// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: BestSellers #render should match snapshot 1`] = `
<section
  className="root page-gutter"
>
  <div
    className="products-wrapper page-max-width"
  >
    <ProductCard
      className="best-seller-card"
      image={
        Object {
          "alt": "Hiphugger",
          "src": "https://thinx-assets.s3.amazonaws.com/best_seller_1.jpg",
        }
      }
      key="Shop Hiphugger"
      product_subtitle="heavy days"
      product_title="Shop Hiphugger"
      url=""
    />
    <ProductCard
      className="best-seller-card"
      image={
        Object {
          "alt": "Cheeky",
          "src": "https://thinx-assets.s3.amazonaws.com/best_seller_2.jpg",
        }
      }
      key="Shop Cheeky"
      product_subtitle="light days"
      product_title="Shop Cheeky"
      url=""
    />
    <ProductCard
      className="best-seller-card"
      image={
        Object {
          "alt": "Thong",
          "src": "https://thinx-assets.s3.amazonaws.com/best_seller_3.jpg",
        }
      }
      key="Shop Thong"
      product_subtitle="lightest days"
      product_title="Shop Thong"
      url=""
    />
    <ProductCard
      className="best-seller-card"
      image={
        Object {
          "alt": "Sport",
          "src": "https://thinx-assets.s3.amazonaws.com/best_seller_4.jpg",
        }
      }
      key="Shop Sport"
      product_subtitle="medium days"
      product_title="Shop Sport"
      url=""
    />
    <ProductCard
      className="best-seller-card"
      image={
        Object {
          "alt": "Boyshort",
          "src": "https://thinx-assets.s3.amazonaws.com/best_seller_5.jpg",
        }
      }
      key="Shop Boyshort"
      product_subtitle="light days"
      product_title="Shop Boyshort"
      url=""
    />
  </div>
</section>
`;
